src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,73 @@
+package duke.task;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import duke.DukeException;
+
+/**
+ * Encapsulation of a deadline.
+ *
+ * @author Sun Ruoxin
+ */
+public class Deadline extends Task {
+    /** The date of the deadline. */
+    protected LocalDate by;
```

Comment:
I think having the variable name be `deadline` or `deadlineDate` would be more descriptive than `by`.